[DON'T MERGE] CASTing a DATE literal without time part to TIME datatype fails

bsatarupa opened the pull request:
### Description
For time datatype PG considers only date value(ex.- '2012-02-23') as bad i/p format. so, above testcase is throwing error from engine code time_in() much before coming to TDS side. 

However, SQL Server allows casting a date literal string to a TIME datatype when the time component in the string is missing: o/p becomes 00:00:00.0000000 

Fix is to add change in time_in() so that it considers date format with missing time info like above one, as valid input when the dialect is TSQL.

--sql_server:
1> select CAST('2012-02-23' AS time)
2> go
----------------------
 00:00:00.0000000

-- Babelfish(before fix)
1> select CAST('2012-02-23' AS time)
2> go
Msg 117440642, Level 16, State 1, Server BABEL, Line 1
invalid input syntax for type time: “2012-02-23"

Task: BABEL-1528
Signed-off-by: Satarupa Biswas [satarupb@amazon.com](mailto:satarupb@amazon.com)

### Test Scenarios Covered ###
o Use case based: Original issue
o Null testcases: N/A
o Negative test scenarios: N/A
o Boundary conditions, Arbitrary inputs: N/A
o Testcases calling from procedures, views and functions
o Minor version upgrade tests, Major version upgrade tests : Done
o Performance tests, Tooling impact, Client tests: N/A


### Check List
- [x] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is under the terms of the Apache 2.0 and PostgreSQL licenses, and grant any person obtaining a copy of the contribution permission to relicense all or a portion of my contribution to the PostgreSQL License solely to contribute all or a portion of my contribution to the PostgreSQL open source project.

For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/babelfish-for-postgresql/babelfish_extensions/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

surendravishnoi commented on test/odbc/psqlodbc/test/time.cpp:
```diff
@@ -139,12 +139,23 @@ TEST_F(PSQL_DataTypes_Time, Insertion_Fail) {
+    "01.00.01",       // mm.yy.dd
+    "19 2000 JAN",    // dd yyyy MONTH
+    "2000 JAN",       // yyyy MONTH
+    // "JAN 19,2000",    // MONTH dd,yyyy *BBF Valid, PG Invalid
```
why are few of these commented?

surendravishnoi commented on test/odbc/psqlodbc/test/time.cpp:
```diff
@@ -139,12 +139,23 @@ TEST_F(PSQL_DataTypes_Time, Insertion_Fail) {
+    "00:00:00.00 AM",
+    "00AM",
+    "00 AM",
+    // "1/31/00",        // m/dd/yy *BBF Valid on Ubuntu 20.x, Invalid on Ubuntu 22.x & PG Ubuntu 20.x
```
why is behaviour different on Ubuntu 22.x? Does same apply to SQL server as well

surendravishnoi commented on test/JDBC/input/BABEL-1528-vu-verify.sql:
```diff
@@ -0,0 +1,14 @@
+select CAST('2012-02-23' AS time)
+GO
+
+select * from BABEL_1528_vu_prepare_t1
```
what is the result if we run this query from PSQL client?

Commit:
CASTing a DATE literal without time part to TIME datatype fails

For time datatype PG considers only date value(ex.- '2012-02-23') as bad i/p format. so, above testcase is throwing error from engine code time_in() much before coming to TDS side.
However, SQL Server allows casting a date literal string to a TIME datatype when the time component in the string is missing: o/p becomes 00:00:00.0000000
Fix is to add change in time_in() so that it considers date format with missing time info like above one, as valid input when the dialect is TSQL.

Task: BABEL-1528
Signed-off-by: Satarupa Biswas satarupb@amazon.com
